operator/v1/types_ingress.go

```diff
@@ -513,6 +520,61 @@ type IngressControllerLogging struct {
 	Access *AccessLogging `json:"access,omitempty"`
 }
 
+// IngressControllerHTTPHeaderPolicy is a policy for setting HTTP headers.
+//
+// +kubebuilder:validation:Enum=Append;Replace;IfNone;Never
+type IngressControllerHTTPHeaderPolicy string
+
+const (
+	// AppendHTTPHeaderPolicy appends the header, preserving any existing header.
+	AppendHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Append"
+	// ReplaceHTTPHeaderPolicy sets the header, removing any existing header.
+	ReplaceHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Replace"
+	// IfNoneHTTPHeaderPolicy sets the header if it is not already set.
+	IfNoneHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "IfNone"
+	// NeverHTTPHeaderPolicy never sets the header, preserving any existing
+	// header.
+	NeverHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Never"
+)
+
+// IngressControllerHTTPForwardedHeaderPolicy describes how the HTTP Forwarded
+// header and related headers are handled.
+type IngressControllerHTTPForwardedHeaderPolicy struct {
+	// policy specifies when and how the IngressController sets the
+	// Forwarded, X-Forwarded-For, X-Forwarded-Host, X-Forwarded-Port,
+	// X-Forwarded-Proto, and X-Forwarded-Proto-Version headers.  The value
+	// may be one of the following:
+	//
+	// * "Append", which specifies that the IngressController appends the
+	//   headers, preserving existing headers.
+	//
+	// * "Replace", which specifies that the IngressController sets the
+	//   headers, replacing any existing Forwarded or X-Forwarded-* headers.
+	//
+	// * "IfNone", which specifies that the IngressController sets the
+	//   headers if they are not already set.
+	//
+	// * "Never", which specifies that the IngressController never sets the
+	//   headers, preserving any existing headers.
+	//
+	// By default, the policy is "Append".
+	//
+	// +optional
+	Policy IngressControllerHTTPHeaderPolicy `json:"policy,omitempty"`
+}
+
+// IngressControllerHTTPHeaders specifies how the IngressController handles
+// certain HTTP headers.
+type IngressControllerHTTPHeaders struct {
+	// forwarded describes how the Forwarded HTTP header and related headers
+	// are handled.
+	//
+	// If this field is empty, the default values are used.
+	//
+	// +optional
+	Forwarded *IngressControllerHTTPForwardedHeaderPolicy `json:"forwarded,omitempty"`
```

Comment:
pointers to strings are a no-no in our configuration.  Empty has a defined meaning and valid, and should mean the same as nil, so nil shouldn't exist.

Reply:
`IngressControllerHTTPForwardedHeaderPolicy` (the type for `spec.httpHeaders.forwarded`) is not a string type.  Are you referring to the fact that `IngressControllerHTTPHeaderPolicy` (the type for `spec.httpHeaders.forwarded.policy`) is a string type?  I think we can make it required but allow empty to avoid breaking upgrades.

Reply:
I changed `Forwarded` to a non-pointer value.  I assume the `+optional` still belongs.  Everything look all right?